swri_roscpp/include/swri_roscpp/subscriber_impl.h

```diff
@@ -81,7 +81,8 @@ class SubscriberImpl
         } else {
           min_latency_ = std::min(min_latency_, latency);
           max_latency_ = std::max(max_latency_, latency);
-          total_latency_ += latency;
+          double latency_weight_ = 0.1;
+          total_latency_.fromSec(latency_weight_*latency.toSec() + (1-latency_weight_)*total_latency_.toSec());
```

Comment:
Actually, it would probably be good to change that 1 to a 1.0 to make sure it is a double

Reply:
Oh no, I was slightly too fast!

Reply:
You're just too responsive!